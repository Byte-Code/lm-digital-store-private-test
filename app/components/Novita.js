import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  height: 60px;
  position: relative;
  z-index: 2;
`;

const TopLabel = styled.div`
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 60px 60px 0 0;
  border-color: #6699cc transparent transparent transparent;
  position: absolute;
  top: 0;
  left: 0;
`;

const Text = styled.span`
 position: absolute;
 width: 85px;
 height: 38px;
 top: -5px;
 left: -26px;
 line-height: 65px;
 text-align: center;
 color: #ffffff;
 text-transform: uppercase;
 font-size: 12px;
 font-family: "LeroyMerlinSansBold", sans-serif;
 -webkit-transform: rotate(316deg); //fai 315deg
 -moz-transform: rotate(316deg);
 -o-transform: rotate(316deg);
 writing-mode: lr-tb;
`;

export default function Icon() {
  return (
    <Wrapper>
      <TopLabel />
      <Text>NOVITÃ€</Text>
    </Wrapper>
  );
}
