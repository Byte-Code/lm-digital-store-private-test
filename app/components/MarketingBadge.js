import React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import styled from 'styled-components';

import Novita from './Novita';

import PrezzoGiuBadge from '../assets/product_badge/abbiamo_abbassato_prezzi.png';
import PromoWebBadge from '../assets/product_badge/PROMO_WEB-2.png';
import PrezzoVincenteBadge from '../assets/product_badge/PREZZO_VINCENTE-2.png';
import DestockBadge from '../assets/product_badge/DESTOCK-2.png';
import PrezzoStockBadge from '../assets/product_badge/PREZZO_STOCK-2.png';
import IdeaPiuBadge from '../assets/product_badge/badge-doppio.png';

const Badge = styled.img`
  width: ${props => (props.width ? props.width : 'auto')};
  height: ${props => (props.height ? props.height : 'auto')};
  position: relative;
  margin-bottom: ${props => props.mBottom || 0}
`;

const Label = styled.p`
  position: absolute;
  bottom: 2px;
  height: 20px;
  width: 100px;
  font-size: 14px;
  background-color: #cc0000;
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  text-transform: uppercase;
`;

export const PrezzoGiu = ({ src }) => (
  <Badge
    height="100px"
    width="100px"
    src={src}
  />
);

export const PromoWeb = ({ src }) => (
  <Badge
    width="100px"
    src={src}
  />
);

export const Destock = ({ src }) => (
  <Badge
    src={src}
  />
);

export const PrezzoVincente = ({ src }) => (
  <Badge
    height="20px"
    width="120px"
    src={src}
  />
);

export const PrezzoStock = ({ src }) => (
  <Badge
    height="20px"
    width="120px"
    src={src}
  />
);

export const IdeaPiu = ({ src, value }) => (
  <div style={{ position: 'relative' }}>
    <Badge
      height="50px"
      width="100px"
      src={src}
      mBottom="18px"
    />
    <Label>{`-${value}% IDEAPIÃš`}</Label>
  </div>
);

const MarketingBadge = ({ promotion }) => {
  switch (promotion.get('code')) {
    case 'PREZZO_GIU':
      return (
        <PrezzoGiu src={PrezzoGiuBadge} />
      );
    case 'NOVITA':
      return (
        <Novita />
      );
    case 'PROMO_WEB':
      return (
        <PromoWeb src={PromoWebBadge} />
      );
    case 'DESTOCK':
      return (
        <Destock src={DestockBadge} />
      );
    case 'PREZZO_STOCK':
      return (
        <PrezzoStock src={PrezzoStockBadge} />
      );
    case 'PREZZO_VINCENTE':
      return (
        <Badge src={PrezzoVincenteBadge} />
      );
    case 'IDEAPIU':
      return (
        <IdeaPiu src={IdeaPiuBadge} value={promotion.get('value')} />
      );
    default:
      return null;
  }
};

MarketingBadge.propTypes = {
  promotion: ImmutablePropTypes.map.isRequired
};

export default MarketingBadge;
