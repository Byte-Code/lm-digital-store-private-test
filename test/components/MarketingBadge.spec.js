import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import MarketingBadge, * as badges from '../../app/components/MarketingBadge';
import Novita from '../../app/components/Novita';


describe('MarketingBadge', () => {
  it('should render null if no promotion is provided', () => {
    const promotion = fromJS({ code: '' });
    const result = shallow(
      <MarketingBadge promotion={promotion} />
    );
    expect(result).toMatchSnapshot();
  });

  it('should render PREZZO_GIU', () => {
    const promotion = fromJS({ code: 'PREZZO_GIU' });
    const result = shallow(
      <MarketingBadge promotion={promotion} />
    );
    const badge = result.find(badges.PrezzoGiu);
    expect(result).toMatchSnapshot();
    expect(badge).toBeDefined();
  });

  it('should render NOVITA', () => {
    const promotion = fromJS({ code: 'NOVITA' });
    const result = shallow(
      <MarketingBadge promotion={promotion} />
    );
    const badge = result.find(Novita);
    expect(result).toMatchSnapshot();
    expect(badge).toBeDefined();
  });

  it('should render PROMO_WEB', () => {
    const promotion = fromJS({ code: 'PROMO_WEB' });
    const result = shallow(
      <MarketingBadge promotion={promotion} />
    );
    const badge = result.find(badges.PromoWeb);
    expect(result).toMatchSnapshot();
    expect(badge).toBeDefined();
  });

  it('should render DESTOCK', () => {
    const promotion = fromJS({ code: 'DESTOCK' });
    const result = shallow(
      <MarketingBadge promotion={promotion} />
    );
    const badge = result.find(badges.Destock);
    expect(result).toMatchSnapshot();
    expect(badge).toBeDefined();
  });

  it('should render PREZZO_STOCK', () => {
    const promotion = fromJS({ code: 'PREZZO_STOCK' });
    const result = shallow(
      <MarketingBadge promotion={promotion} />
    );
    const badge = result.find(badges.PrezzoStock);
    expect(result).toMatchSnapshot();
    expect(badge).toBeDefined();
  });

  it('should render PREZZO_VINCENTE', () => {
    const promotion = fromJS({ code: 'PREZZO_VINCENTE' });
    const result = shallow(
      <MarketingBadge promotion={promotion} />
    );
    const badge = result.find(badges.PrezzoVincente);
    expect(result).toMatchSnapshot();
    expect(badge).toBeDefined();
  });

  it('should render IDEAPIU', () => {
    const promotion = fromJS({ code: 'IDEAPIU', value: 10 });
    const result = shallow(
      <MarketingBadge promotion={promotion} />
    );
    const badge = result.find(badges.IdeaPiu);
    expect(result).toMatchSnapshot();
    expect(badge).toBeDefined();
  });
});
