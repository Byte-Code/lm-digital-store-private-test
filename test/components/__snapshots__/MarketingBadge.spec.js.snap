// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketingBadge should render DESTOCK 1`] = `
<MarketingBadge
  promotion={
    Object {
      "code": "DESTOCK",
    }
  }
>
  <Component
    src="test-file-stub"
  >
    <styled.img
      src="test-file-stub"
    >
      <img
        className="cuDvmw"
        src="test-file-stub"
      />
    </styled.img>
  </Component>
</MarketingBadge>
`;

exports[`MarketingBadge should render IDEAPIU 1`] = `
<MarketingBadge
  promotion={
    Object {
      "code": "IDEAPIU",
      "value": 10,
    }
  }
>
  <Component
    src="test-file-stub"
    value={10}
  >
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <styled.img
        height="50px"
        mBottom="18px"
        src="test-file-stub"
        width="100px"
      >
        <img
          className="gJrHnD"
          height="50px"
          src="test-file-stub"
          width="100px"
        />
      </styled.img>
      <styled.p>
        <p
          className="bXfyzC"
        >
          -10% IDEAPIÚ
        </p>
      </styled.p>
    </div>
  </Component>
</MarketingBadge>
`;

exports[`MarketingBadge should render NOVITA 1`] = `
<MarketingBadge
  promotion={
    Object {
      "code": "NOVITA",
    }
  }
>
  <Icon>
    <styled.div>
      <div
        className="bjsMfp"
      >
        <styled.div>
          <div
            className="gZQECl"
          />
        </styled.div>
        <styled.span>
          <span
            className="hDeyYv"
          >
            NOVITÀ
          </span>
        </styled.span>
      </div>
    </styled.div>
  </Icon>
</MarketingBadge>
`;

exports[`MarketingBadge should render PREZZO_GIU 1`] = `
<MarketingBadge
  promotion={
    Object {
      "code": "PREZZO_GIU",
    }
  }
>
  <Component
    src="test-file-stub"
  >
    <styled.img
      height="100px"
      src="test-file-stub"
      width="100px"
    >
      <img
        className="jqBbJk"
        height="100px"
        src="test-file-stub"
        width="100px"
      />
    </styled.img>
  </Component>
</MarketingBadge>
`;

exports[`MarketingBadge should render PREZZO_STOCK 1`] = `
<MarketingBadge
  promotion={
    Object {
      "code": "PREZZO_STOCK",
    }
  }
>
  <Component
    src="test-file-stub"
  >
    <styled.img
      height="20px"
      src="test-file-stub"
      width="120px"
    >
      <img
        className="fUkjDv"
        height="20px"
        src="test-file-stub"
        width="120px"
      />
    </styled.img>
  </Component>
</MarketingBadge>
`;

exports[`MarketingBadge should render PREZZO_VINCENTE 1`] = `
<MarketingBadge
  promotion={
    Object {
      "code": "PREZZO_VINCENTE",
    }
  }
>
  <styled.img
    src="test-file-stub"
  >
    <img
      className="cuDvmw"
      src="test-file-stub"
    />
  </styled.img>
</MarketingBadge>
`;

exports[`MarketingBadge should render PROMO_WEB 1`] = `
<MarketingBadge
  promotion={
    Object {
      "code": "PROMO_WEB",
    }
  }
>
  <Component
    src="test-file-stub"
  >
    <styled.img
      src="test-file-stub"
      width="100px"
    >
      <img
        className="GjBEV"
        src="test-file-stub"
        width="100px"
      />
    </styled.img>
  </Component>
</MarketingBadge>
`;

exports[`MarketingBadge should render null if no promotion is provided 1`] = `
<MarketingBadge
  promotion={
    Object {
      "code": "",
    }
  }
/>
`;
